roteiro-de-estudos/paradigmas/cp-problem-explanation.md

```diff
@@ -0,0 +1,151 @@
+# O que é Programação Competitiva?
+
+Inicialmente, é necessário que não restem dúvidas para o leitor acerca das definições que estão sendo empregadas. Dessa forma, é possível descrever Programação Competitiva a partir da seguinte definição
+
+> *“Competitive programming is solving well-defined problems by writing computer programs under specified limits”*
+
+A definição acima pode ser livremente traduzida para o Português da seguinte maneira
+
+> *“Programação Competitiva trata-se da resolução de problemas bem definidos mediante a escrita de programas de computador dentro de limites especificados"*
+
+A partir da definição de Programação Competitiva, é possível ainda descrever com mais detalhes alguns termos presentes na definição, como *problemas bem definidos* e *programas de computador dentro de limites especificados*.
+
+O termo *problemas bem definidos*, indica que cada problema de Programação Competitiva possuirá uma instrução clara e objetiva de como as informações descritas no problema irão se relacionar entre si a fim de gerar a resposta esperada, sendo assim, existirá somente um único conjunto de respostas correspondente ao que é pedido no problema. Uma outra forma de entender o trecho é que um problema de programação competitiva não abre margem para o leitor definir quais seriam as relações entre as informações descritas no problema que são consideradas aceitáveis.
```

Comment:
Acho que ficou difícil de entender a definição do tempo *problema bem definido*.

Sugestão:
_Well-defined (well-structured) problems are those that contain a clear specification of three elements of the problem space: the initial state (the problem situation), the set of operators (rules and strategies) to solve the problem, and the goal state (the solution)_